lib/cucumber/glue/hook.rb

```diff
@@ -1,14 +1,17 @@
 # frozen_string_literal: true
 
+require 'securerandom'
+require 'cucumber/messages'
 require 'cucumber/glue/invoke_in_world'
 
 module Cucumber
   module Glue
     # TODO: Kill pointless wrapper for Before, After and AfterStep hooks with fire
     class Hook
-      attr_reader :tag_expressions, :location
+      attr_reader :id, :tag_expressions, :location
 
       def initialize(registry, tag_expressions, proc)
+        @id = SecureRandom.uuid
```

Comment:
We should use `IdGenerator` here, so we can plug in the predictable one. This will be necessary for the cucumber compatibility kit, which will compare generated and expected ndjson.

I think it's best to pass the `id` through the constructor, and let the caller delegate to the `IdGenerator` to generate it. The generator type should be configurable via a `--predictable-ids` command-line option.